java/lang/sqli.yml

```diff
@@ -0,0 +1,44 @@
+patterns:
+  - pattern: |
+      $<_>.$<METHOD>($<DYNAMIC_INPUT_LITERAL>)
+    filters:
+      - variable: METHOD
+        values:
+          - execute
+          - executeQuery
+          - prepareStatement
+      - variable: DYNAMIC_INPUT_LITERAL
+        detection: java_lang_sqli_dynamic_input
+auxiliary:
+  - id: java_lang_sqli_dynamic_input
+    patterns:
+      - pattern: |
+          $<STRING_LITERAL> + $<DYNAMIC_INPUT>
+        filters:
+          - variable: STRING_LITERAL
+            detection: string_literal
+            contains: false
+          - not:
+              variable: DYNAMIC_INPUT
+              detection: string_literal
+              contains: false
+languages:
+  - java
+trigger:
+  match_on: presence
+metadata:
+  description: "SQL injection threat detected"
+  remediation_message: |
+    Instead of using dynamically crafted string for your sql query use prepared statements instead
+
+    PreparedStatement
```

Comment:
Should this line be removed?

Maybe you could give a bad example as well.

Reply:
also the bad examples are in testdata, not sure if it even makes sense to duplicate it?

Reply:
The testdata is for checking the rule is working. Whereas the remediation message is to inform the user. So I think it's useful to include one example there.